### PR TITLE
Create a tabular structure to display data and some actions to be performed #7

### PR DESCRIPTION
**Pull request by ystanwar**

### CODE REVIEW

**Comment by ystanwar** [comment]
> Please write tests using React Testing Library

ok, will do the same

File: ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import './AllAppointmentServices.scss'
```
Comment (vinay-ThoughtWorks) [inline ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx]: Prefer to use css modules and classnames. That generates unique classnames after compilation which prevents from conflicting with similar class names in other modules/components.

File: ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                    {this.getHeader()}
+                    </tr>  
+                </thead>
+                <tbody>
+                    {this.getRowsData()}
+                </tbody>
+            </table>
+        </div>
+    }
+};
```
[the remark this answers is not in the document]
Comment (ystanwar) [reply ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx]: used hooks, although I was not maintaining state, so just removed class component with function and made other supporting changes

File: ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        let keys = Object.keys(this.props.services[0]);
+        keys.push("Actions");
+        return keys;
+    }
+
+    getHeader = function () {
+        var keys = this.getKeys();
+        return keys.map((key,index)=>{
+                return <th key ={index}>{key}</th>
+        })
```
[the remark this answers is not in the document]
Comment (ystanwar) [reply ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx]: removed extra return

File: ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        this.getKeys =this.getKeys.bind(this);
+        this.getHeader =this.getHeader.bind(this);
+        this.getRowsData =this.getRowsData.bind(this);
+    }
+
+    getKeys = function () {
+        let keys = Object.keys(this.props.services[0]);
+        keys.push("Actions");
+        return keys;
+    }
```
[the remark this answers is not in the document]
Comment (ystanwar) [reply ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx]: done

File: ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx (hunk 1 above)
Comment (ystanwar) [reply ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx]: looking into it

File: ui/react-components/components/AllAppointmentServices/AllAppointmentServices.scss [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+table td a {
+    display: inline-block;
+    margin: 0 5px;
+}
+
+a, a:active, a:hover, a:link, a:visited {
+    color: #2b95ff;
+    text-decoration: none;
+    cursor: pointer;
+}
```
[the remark this answers is not in the document]
Comment (ystanwar) [reply ui/react-components/components/AllAppointmentServices/AllAppointmentServices.scss]: looking into it

File: ui/react-components/components/AllAppointmentServices/AllAppointmentServices.test.jsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { shallow } from 'enzyme';
```
[the remark this answers is not in the document]
Comment (ystanwar) [reply ui/react-components/components/AllAppointmentServices/AllAppointmentServices.test.jsx]: used React Testing Library , but facing some issues

File: ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (vinay-ThoughtWorks) [reply ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx]: Go through this link once. For better usage and understanding of map function. Also update the required changes in all remaining in you code.

File: ui/react-components/components/AllAppointmentServices/AllAppointmentServices.test.jsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (ystanwar) [reply ui/react-components/components/AllAppointmentServices/AllAppointmentServices.test.jsx]: done

File: ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ystanwar) [reply ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx]: done

File: ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+    getHeader = function () {
+        var keys = this.getKeys();
```
[the remark this answers is not in the document]
Comment (ystanwar) [reply ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx]: done

File: ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (ystanwar) [reply ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx]: done

File: ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx (hunk 1 above)
Comment (ystanwar) [reply ui/react-components/components/AllAppointmentServices/AllAppointmentServices.jsx]: done

File: ui/react-components/components/AllAppointmentServices/AllAppointmentServices.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (ystanwar) [reply ui/react-components/components/AllAppointmentServices/AllAppointmentServices.scss]: done

**Comment by bsneha90** [comment]
Closing this as Neha and Yuraj are going to raise a different PR

**Pull request closed without merging**